state/model.go

```diff
@@ -83,6 +83,51 @@ type modelDoc struct {
 	// LatestAvailableTools is a string representing the newest version
 	// found while checking streams for new versions.
 	LatestAvailableTools string `bson:"available-tools,omitempty"`
+
+	// SLA is the current support level of the model.
+	SLA slaDoc `bson:"sla"`
+}
+
+// slaLevel enumerates the support levels available to a model.
+type slaLevel string
```

Comment:
what is the value of defining a new type? if we want to make sure the SLA is one of the four (currently) supported strings, that can be checked in a single function and we can remove a bit of code here..